### PR TITLE
RESOURCE-123 Support servicebus subscription rules

### LINKED ISSUE #558: Service Bus Rules

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

Support Service Bus Subscription Rule (s)

### Issues Resolved
#558 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by soumyo13 — CHANGES_REQUESTED** [review]
Please remove the "azure_service_bus_namespace" and "azure_service_bus_regions" resource files from this PR. both are separate PR.

File: docs/resources/azure_service_bus_subscription_rule.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure).
+For an example `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+`resource_group`, `namespace_name`, `subscription_name`, `topic_name` and `name` are required parameters.
```
Comment (soumyo13) [inline docs/resources/azure_service_bus_subscription_rule.md:29]:
```suggestion
`resource_group`, `namespace_name`, `subscription_name`, `topic_name`, and `name` are required parameters.
```

File: libraries/azure_service_bus_subscription_rule.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'azure_generic_resource'
+
+class AzureServiceBusSubscriptionRule < AzureGenericResource
+  name 'azure_service_bus_subscription_rule'
+  desc 'Retrieves and verifies the settings of an Azure Service Bus Subscription Rule.'
+  example <<-EXAMPLE
+    describe azure_service_bus_subscription_rule(resource_group: 'inspec-rg', namespace_name: 'inspec-ns', topic_name: 'inspec-topic', subscription_name: 'inspec-sub', name: 'inspec_rule1') do
```
Comment (soumyo13) [inline libraries/azure_service_bus_subscription_rule.rb:7]:
```suggestion
    describe azure_service_bus_subscription_rule(resource_group: 'RESOURCE_GROUP', namespace_name: 'NAMESPACE_NAME', subscription_name: "SUBSCRIPTION_NAME", topic_name: 'TOPIC_NAME', name: 'SUBSCRIPTION_RULE_NAME') do
```

File: libraries/azure_service_bus_subscription_rule.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      it { should exist }
+    end
+  EXAMPLE
+
+  def initialize(opts = {})
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.ServiceBus/namespaces', opts)
+    opts[:required_parameters] = %i(namespace_name topic_name subscription_name)
```
Comment (soumyo13) [inline libraries/azure_service_bus_subscription_rule.rb:16]: Should we provide "resource_group" in the validate parameters?

File: libraries/azure_service_bus_subscription_rule.rb (hunk 2 above)
Comment (soumyo13) [inline libraries/azure_service_bus_subscription_rule.rb:7]:
```suggestion
    describe azure_service_bus_subscription_rule(resource_group: 'RESOURCE_GROUP', namespace_name: 'NAMESPACE_NAME', subscription_name: "SUBSCRIPTION_NAME", topic_name: 'TOPIC_NAME', name: 'SUBSCRIPTION_RULE_NAME') do
```

File: libraries/azure_service_bus_subscription_rules.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+require 'azure_generic_resources'
+
+class AzureServiceBusSubscriptionRules < AzureGenericResources
+  name 'azure_service_bus_subscription_rules'
+  desc 'Verifies settings for a collection of Azure Service Bus Subscription Rules in a Resource Group.'
+  example <<-EXAMPLE
+    describe azure_service_bus_subscription_rules(resource_group: 'inspec-rg', namespace_name: 'inspec-ns', subscription_name: 'inspec-subs', topic_name: 'inspec-topic') do
```
Comment (soumyo13) [inline libraries/azure_service_bus_subscription_rules.rb:7]:
```suggestion
    describe azure_service_bus_subscription_rules(resource_group: 'RESOURCE_GROUP', namespace_name: 'NAMESPACE_NAME', subscription_name: 'SUBSCRIPTION_NAME', topic_name: 'TOPIC_NAME') do
```

**Review by soumyo13 — CHANGES_REQUESTED** [review]

File: libraries/azure_service_bus_subscription_rule.rb (hunk 3 above)
Comment (sathish-progress) [reply libraries/azure_service_bus_subscription_rule.rb:16]: handled implicitly here -> https://github.com/inspec/inspec-azure/blob/main/libraries/azure_backend.rb#L185

**Review by soumyo13 — APPROVED** [review]

**Comment by IanMadd** [comment]
I deleted the other markdown pages. Let me know if it's wrong and I can undo it.

**Review by IanMadd — APPROVED** [review]

**Pull request merged**